src/Storages/MergeTree/IMergeTreeDataPart.h

```diff
@@ -170,7 +170,8 @@ class IMergeTreeDataPart : public std::enable_shared_from_this<IMergeTreeDataPar
     bool isEmpty() const { return rows_count == 0; }
 
     /// Compute part block id for zero level part. Otherwise throws an exception.
-    String getZeroLevelPartBlockID() const;
+    /// If token is not empty, block id is calculated based on it instead of block data
+    String getZeroLevelPartBlockID(const String & token = String()) const;
```

Comment:
Default value is unnecessary.

Reply:
Yes, but after deduplication token will be added to non-replicated MergeTree  as well